server/authorization.go

```diff
@@ -19,23 +19,26 @@ func (ae *authError) Error() string {
 	return string(errorString)
 }
 
-var oAuthMessage string = "[Click here to link your Microsoft account.](%s/plugins/" + manifest.Id + "/oauth2/connect?channelID=%s)"
+func (p *Plugin) getOauthMessage(channelID string) (string, error) {
+	pluginOauthURL, err := p.getPluginOauthURL()
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("[Click here to link your Microsoft account.](%s/connect?channelID=%s)", pluginOauthURL, channelID), nil
```

Comment:
`channelID` needs to be query escaped
https://pkg.go.dev/net/url#QueryEscape

Reply:
@enzowritescode Added, please re-review